presto-thrift-testing-server/src/main/java/com/facebook/presto/connector/thrift/server/ThriftIndexedTpchService.java

```diff
@@ -0,0 +1,205 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.connector.thrift.server;
+
+import com.facebook.presto.connector.thrift.api.PrestoThriftBlock;
+import com.facebook.presto.connector.thrift.api.PrestoThriftId;
+import com.facebook.presto.connector.thrift.api.PrestoThriftNullableToken;
+import com.facebook.presto.connector.thrift.api.PrestoThriftPageResult;
+import com.facebook.presto.connector.thrift.api.PrestoThriftSchemaTableName;
+import com.facebook.presto.connector.thrift.api.PrestoThriftServiceException;
+import com.facebook.presto.connector.thrift.api.PrestoThriftSplit;
+import com.facebook.presto.connector.thrift.api.PrestoThriftSplitBatch;
+import com.facebook.presto.spi.ConnectorPageSource;
+import com.facebook.presto.spi.RecordPageSource;
+import com.facebook.presto.spi.RecordSet;
+import com.facebook.presto.spi.type.Type;
+import com.facebook.presto.split.MappedRecordSet;
+import com.facebook.presto.tests.tpch.TpchIndexedData;
+import com.facebook.presto.tests.tpch.TpchIndexedData.IndexedTable;
+import com.facebook.presto.tests.tpch.TpchScaledTable;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+import static com.facebook.presto.connector.thrift.server.SplitInfo.indexSplit;
+import static com.facebook.presto.tests.AbstractTestIndexedQueries.INDEX_SPEC;
+import static com.google.common.base.Preconditions.checkArgument;
+
+public class ThriftIndexedTpchService
+        extends ThriftTpchService
+{
+    private static final int NUMBER_OF_INDEX_SPLITS = 2;
+    private final TpchIndexedData indexedData = new TpchIndexedData("tpchindexed", INDEX_SPEC);
+
+    @Override
+    protected List<Set<String>> getIndexableKeys(String schemaName, String tableName)
+    {
+        TpchScaledTable tpchScaledTable = new TpchScaledTable(tableName, schemaNameToScaleFactor(schemaName));
+        return ImmutableList.copyOf(INDEX_SPEC.getColumnIndexes(tpchScaledTable));
+    }
+
+    @Override
+    protected PrestoThriftSplitBatch getLookupSplitsSync(
+            PrestoThriftSchemaTableName schemaTableName,
+            List<String> lookupColumnNames,
+            PrestoThriftPageResult keys,
+            int maxSplitCount,
+            PrestoThriftNullableToken nextToken)
+            throws PrestoThriftServiceException
+    {
+        checkArgument(NUMBER_OF_INDEX_SPLITS <= maxSplitCount, "maxSplitCount for lookup splits is too low");
```

Comment:
Technically, this should not happen if the engine is working as planned. In case it happens actual calls for data will fail which I think is good enough.